File: Java/Crud.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Iterator;
// import Mhs.java;

class Crud extends Mhs {
    // atribut
    private String nama, nim, prodi, fakultas;
    ArrayList<Mhs> dataMhs = new ArrayList<>();

    // constructor
    public Crud() {}

    // method untuk create update delete
    public void Create() {
        Scanner sc = new Scanner(System.in);

        System.out.println("Masukkan data mahasiswa :");
        System.out.print("Nama : ");
        if (sc.hasNext()) nama = sc.next();
        System.out.print("NIM : ");
        if (sc.hasNext()) nim = sc.next();
        System.out.print("Program Studi : ");
        if (sc.hasNext()) prodi = sc.next();
        System.out.print("Fakultas : ");
        if (sc.hasNext()) fakultas = sc.next();

        Mhs temp = new Mhs(nama, nim, prodi, fakultas);
        dataMhs.add(temp);

        System.out.println("Data berhasil ditambahkan\n");
        sc.close();
    }

    public void Update() {
        Scanner sc = new Scanner(System.in);

        System.out.print("Masukkan NIM mahasiswa yang mau diupdate : ");
        nim = sc.next();

        int x = 0;
        for (int i = 0; i < dataMhs.size(); i++) {
            if (dataMhs.get(i).getNim() == nim) {
                x = 1;

                // nerima input yang baru
                System.out.println("Masukkan data mahasiswa terbaru :");
                System.out.print("Nama : ");
                if (sc.hasNext()) nama = sc.next();
                System.out.print("NIM : ");
                if (sc.hasNext()) nim = sc.next();
                System.out.print("Program Studi : ");
                if (sc.hasNext()) prodi = sc.next();
                System.out.print("Fakultas : ");
                if (sc.hasNext()) fakultas = sc.next();

                // ubah yang sekarang
                dataMhs.get(i).setNama(nama);
                dataMhs.get(i).setNim(nim);
                dataMhs.get(i).setProdi(prodi);
                dataMhs.get(i).setFakultas(fakultas);
            }
        }

        if (x == 0) {
            System.out.println("NIM tidak ditemukan");
        } else {
            System.out.println("Data berhasil diubah");
        }

        sc.close();
    }

    public void Delete() {
        Scanner sc = new Scanner(System.in);

        System.out.print("Masukkan NIM mahasiswa yang mau dihapus : ");
        nim = sc.next();

        // int x = 0;
        // Iterator itr = dataMhs.iterator();
        // while (itr.hasNext()) {
        //     String temp = itr.getNim();
        //     if (temp == nim) {
        //         x = 1;
        //         dataMhs.remove(itr);
        //     }
        // }

        dataMhs.remove(String.valueOf(nim));

        // if (x == 0) {
        //     System.out.println("NIM tidak ditemukan");
        // } else {
            System.out.println("Data berhasil dihapus");
        // }

        sc.close();
    }

    public void Print() {
        System.out.println("DATA MAHASISWA");
        int itr = 1;

        for (int i = 0; i < dataMhs.size(); i++) {
            System.out.println(itr + ". " + dataMhs.get(i).getNama() + " " +
                                dataMhs.get(i).getNim() + " " +
                                dataMhs.get(i).getProdi() + " " +
                                dataMhs.get(i).getFakultas());
            i++;
        }
    }
}

File: CPP/crud.cpp
#include <iostream>
#include <string>
#include "mhs.cpp"

using namespace std;
#include <list>

class CRUD : public Mhs {
    private:
        list<Mhs> dataMhs; // untuk data mahasiswa
        string nama, nim, prodi, fakultas;

    public:
        CRUD() {}

        void Create() {
            cout << "\nMasukkan data mahasiswa :\n";
            cout << "Nama : ";
            cin >> nama;
            cout << "NIM : ";
            cin >> nim;
            cout << "Program Studi : ";
            cin >> prodi;
            cout << "Fakultas : ";
            cin >> fakultas;

            Mhs temp(nama, nim, prodi, fakultas);

            dataMhs.push_back(temp);

            cout << "Data berhasil ditambahkan\n";
        }

        void Update() {
            cout << "\nMasukkan NIM mahasiswa yang mau diupdate : ";
            cin >> nim;

            int x = 0;
            for (list<Mhs>::iterator it = dataMhs.begin(); it != dataMhs.end(); it++) {
                if (it->getNim() == nim) {
                    x = 1;

                    cout << "\nMasukkan data mahasiswa terbaru : ";
                    cout << "Nama : ";
                    cin >> nama;
                    cout << "NIM : ";
                    cin >> nim;
                    cout << "Program Studi : ";
                    cin >> prodi;
                    cout << "Fakultas : ";
                    cin >> fakultas;

                    // ubah yang sekarang
                    it->setNama(nama);
                    it->setNim(nim);
                    it->setProdi(prodi);
                    it->setFakultas(fakultas);
                }
            }
            if (x == 0) {
                cout << "NIM tidak ditemukan\n";
            } else {
                cout << "Data berhasil diubah\n";
            }
        }

        void Delete() {
            cout << "\nMasukkan NIM mahasiswa yang mau dihapus : ";
            cin >> nim;

            int x = 0;
            list<Mhs>::iterator it = dataMhs.begin();
            while (it != dataMhs.end() && x == 0) {
                if (it->getNim() == nim) {
                    x = 1;
                    dataMhs.erase(it);
                }
                it++;
            }

            if (x == 0) {
                cout << "NIM tidak ditemukan\n";
            } else {
                cout << "Data berhasil dihapus\n";
            }
        }

        void Print() {
            cout << "\nDATA MAHASISWA\n";
            int i = 1;
            
            for (list<Mhs>::iterator it = dataMhs.begin(); it != dataMhs.end(); it++) {
                cout << i << ". " << it->getNama() << " "
                    << it->getNim() << " " << it->getProdi() << " "
                    << it->getFakultas() << "\n";
            }
        }

        ~CRUD() {}
};
